tests/main/interfaces-shared-memory/shm-slot/meta/snap.yaml

```diff
@@ -0,0 +1,12 @@
+name: shm-slot
+version: 1.0
+apps:
+  cmd:
+    command: bin/cmd
+    slots: [shmem]
+slots:
+  shmem:
+    interface: shared-memory
+    shared-memory: super-foo
+    write: [writable-bar]
```

Comment:
could you either add a path that includes a regular expression or modify this one to be a pattern so we can show that the regular expression works too?

Reply:
Uh! This fails, because in the validation code I have a call to `apparmor.ValidateNoAppArmorRegexp()` -- I obviously must not have noticed that part of the spec where you talk about supporting "`*`". Since the handling of it is not trivial (I added a question to the spec document, please have a look), I wonder if we can merge this without it? I added a TODO comment.

Reply:
Ah I see, sorry I keep forgetting that all your code about validating AARE is still not merged yet :-) 

I think it's fine in this case to not support regular expressions in this first iteration in this PR, and then after mount-control step 2 PR lands with the AARE stuff then I think we can tie that validation into this, but to be clear we do definitely need to allow wild-cards specifically in the final version of the interface as that's critical for customer use-cases.